src/homestead_xml_utils.cpp

```diff
@@ -280,6 +303,13 @@ std::vector<std::string> get_public_ids(const std::string& user_data)
   {
     TRC_ERROR("Failed to extract any ServiceProfile/PublicIdentity/Identity nodes from %s", user_data.c_str());
   }
+
+  // Set the default id - this is the first unbarred public identity.
+  if (unbarred_public_ids.size() != 0)
```

Comment:
I would have done `!unbarred_public_ids.empty()` but the existing code does the same thing for `public_ids` just above so I don't really care.